### PR TITLE
Store YouTube Channel Info

### PR DESCRIPTION
**Pull request by TheRealPikachu**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: actions/Store YouTube Channel Info [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  // so be sure to provide checks for variable existance.
+  // ---------------------------------------------------------------------
+
+  action: async function (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const query = this.evalMessage(data.query, cache)
+    const info = parseInt(data.info)
+    const id = query
```
Comment (almeidx) [inline actions/Store YouTube Channel Info:126]: ?

File: actions/Store YouTube Channel Info [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    let result = 7
+    const channelId = id
```
Comment (almeidx) [inline actions/Store YouTube Channel Info:128]: ?

File: actions/Store YouTube Channel Info [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  // ---------------------------------------------------------------------
+
+  action: async function (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const query = this.evalMessage(data.query, cache)
+    const info = parseInt(data.info)
+    const id = query
+    let result = 7
```
Comment (almeidx) [inline actions/Store YouTube Channel Info:127]: ?

File: actions/Store YouTube Channel Info [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    const channelId = id
+    const ytch = require('yt-channel-info')
```
Comment (almeidx) [inline actions/Store YouTube Channel Info:129]: Use Mods.require()

File: actions/Store YouTube Channel Info [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+  // ---------------------------------------------------------------------
+  // Action Subtitle
+  //
+  // This function generates the subtitle displayed next to the name.
+  // ---------------------------------------------------------------------
+
+  subtitle: function (data) {
+    // Each item corresponds to each switch statement.
+    const info = ['Item 1', 'Item 2', 'Item 3']
```
Comment (almeidx) [inline actions/Store YouTube Channel Info:26]: ?

File: actions/Store YouTube Channel Info [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+  name: 'Store YouTube Channel Info',
+
+  // ---------------------------------------------------------------------
+  // Action Section
+  //
+  // This is the section the action will fall into.
+  // ---------------------------------------------------------------------
+
+  section: 'Audio Control',
```
Comment (almeidx) [inline actions/Store YouTube Channel Info:16]: Not directly related with audio

File: actions/store_youtube_channel_info.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  // so be sure to provide checks for variable existance.
+  // ---------------------------------------------------------------------
+
+  action: async function (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const query = this.evalMessage(data.query, cache)
+    const info = parseInt(data.info)
+    const channelId = query
```
Comment (almeidx) [inline actions/store_youtube_channel_info.js:123]: Why?

Comment (TheRealPikachu) [reply actions/store_youtube_channel_info.js:123]: Right, I'll remove query

File: actions/store_youtube_channel_info_MOD.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
+module.exports = {
+
+  name: 'Store YouTube Channel Info',
+
+  section: 'YouTube Tools',
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:5]:
```suggestion
  name: 'Store YouTube Channel Info',
  section: 'YouTube Tools',
```

File: actions/store_youtube_channel_info_MOD.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+  subtitle: function (data) {
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:7]:
```suggestion
  subtitle (data) {
```

File: actions/store_youtube_channel_info_MOD.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    return 'Store information about a YouTube channel.'
+  },
+
+  variableStorage: function (data, varType) {
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:11]:
```suggestion
  variableStorage (data, varType) {
```

File: actions/store_youtube_channel_info_MOD.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    const type = parseInt(data.storage)
+    if (type !== varType) return
+    const dataType = 'YouTube Channel Info'
+    return ([data.varName, dataType])
+  },
+
+  fields: ['query', 'info', 'storage', 'varName'],
+
+  html: function (isEvent, data) {
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:20]:
```suggestion
  html (isEvent, data) {
```

File: actions/store_youtube_channel_info_MOD.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    </select>
+  </div>
+  <div id="varNameContainer" style="float: right; width: 60%;">
+    Variable Name:<br>
+    <input id="varName" class="round" type="text">
+  </div>
+</div>`
+  },
+
+  init: function () {},
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:52]:
```suggestion
  init () {},
```

File: actions/store_youtube_channel_info_MOD.js [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+  action: async function (cache) {
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:54]:
```suggestion
  async action (cache) {
```

File: actions/store_youtube_channel_info_MOD.js [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+      case 6:
+        result = maininfo.isFamilyFriendly
+        break
+    }
+
+    this.storeValue(result, storage, varName, cache)
+    this.callNextAction(cache)
+  },
+
+  mod: function (DBM) {}
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:94]:
```suggestion
  mod () {}
```

**Comment by TheRealPikachu** [comment]
No, I dont think I'm gonna do that

**Comment by TheRealPikachu** [comment]
Someone else already made a mod for that

File: actions/store_youtube_channel_info_MOD.js [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+  async action (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const channelId = this.evalMessage(data.query, cache)
+    const info = parseInt(data.info)
+    const Mods = this.getMods()
+    const ytch = Mods.require('yt-channel-info')
+    let result = 0
```
Comment (almeidx) [inline actions/store_youtube_channel_info_MOD.js:60]:
```suggestion
    let result
```

**Comment by TheRealPikachu** [comment]
what's section:actions supposed to mean

**Pull request merged**